src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentMarkupEncodingPass.cs

```diff
@@ -65,6 +71,145 @@ public override void VisitHtml(HtmlContentIntermediateNode node)
                         }
                     }
                 }
+
+                // If we reach here, we don't have newlines, tabs or non-ascii characters in this node.
+                // If we can successfully decode all HTML entities(if any) in this node, we can safely let it call AddContent.
+                var decodedContent = new List<string>();
```

Comment:
If you wanted to go full #RyanMode you could make this an array since we already know what the size will be.